nav2_collision_monitor/src/polygon.cpp

```diff
@@ -168,8 +169,46 @@ bool Polygon::isShapeSet()
   return true;
 }
 
-void Polygon::updatePolygon()
+bool Polygon::isUsingVelocityPolygonSelector()
 {
+  return !velocity_polygons_.empty();
+}
+
+void Polygon::updatePolygon(const Velocity & cmd_vel_in)
+{
+  if (isUsingVelocityPolygonSelector()) {
+    for (auto & velocity_polygon : velocity_polygons_) {
+
+      if (velocity_polygon->isInRange(cmd_vel_in)) {
+        // Set the polygon that is within the speed range
+        poly_ = velocity_polygon->getPolygon();
+
+        // Update visualization polygon
+        polygon_.polygon.points.clear();
+        for (const Point & p : poly_) {
+          geometry_msgs::msg::Point32 p_s;
+          p_s.x = p.x;
+          p_s.y = p.y;
+          // p_s.z will remain 0.0
+          polygon_.polygon.points.push_back(p_s);
+        }
+
+        return;
```

Comment:
So... what happens if there are multiple polygons that apply to the range potentially?

Reply:
This might be related to https://github.com/ros-planning/navigation2/pull/3708#discussion_r1272210118 thread discussion. If so, I thing we agreed to select first suitable VelocityPolygon in the list and reflect this rule in the documentation